static/js/containers/CreatePostPage_test.js

```diff
@@ -512,5 +507,76 @@ describe("CreatePostPage", () => {
         }
       })
     })
+
+    //
+    ;[
+      [LINK_TYPE_ANY, LINK_TYPE_ANY, LINK_TYPE_LINK, true, false],
+      [LINK_TYPE_ANY, LINK_TYPE_ANY, LINK_TYPE_TEXT, true, false],
+      [LINK_TYPE_ANY, LINK_TYPE_ANY, null, false, false],
+      [LINK_TYPE_LINK, LINK_TYPE_ANY, LINK_TYPE_LINK, true, false],
+      [LINK_TYPE_LINK, LINK_TYPE_ANY, LINK_TYPE_LINK, false, true],
+      [LINK_TYPE_TEXT, LINK_TYPE_ANY, LINK_TYPE_TEXT, true, false],
+      [LINK_TYPE_TEXT, LINK_TYPE_ANY, LINK_TYPE_TEXT, false, true]
+    ].forEach(
+      ([
+        fromChannelType,
+        toChannelType,
+        formValue,
+        hasUrlOrText,
+        shouldDispatch
+      ]) => {
+        it(`${shouldIf(
+          shouldDispatch
+        )} dispatch FORM_UPDATE if the postType is ${String(
+          formValue
+        )} when it goes from ${String(
+          fromChannelType
+        )} to ${toChannelType} and user has ${
+          hasUrlOrText ? "" : "not "
+        }entered a value`, () => {
```

Comment:
Oof... I think I misinterpreted this test before when I suggested we should split them up. This is basically a carbon copy of the test case above with some added , and I now I understand why it needs all of this logic. So going against my ill-conceived suggestion before, this should definitely be just 1 parametrized test case.

I think the description and parameters could be clearer though. `formValue` should be `postType` or `formPostType`, `hasUrlOrText` could be something like `hasInput`, and the description could be changed to something like this:

```javascript
it(`${shouldIf(
          shouldDispatch
        )} reset form if channel type changes from ${String(
          fromChannelType
        )} to ${toChannelType}, postType=${String(
          formPostType
        )}, user input is ${
          hasInput ? "not " : ""
        }empty`
```

Reply:
Test has been updated with new parameters/description